source/Ship.cpp

```diff
@@ -3374,6 +3375,66 @@ double Ship::IdleHeat() const
 	double dissipation = HeatDissipation() + activeCooling / MaximumHeat();
 	if(!dissipation) return production ? numeric_limits<double>::max() : 0;
 	return production / dissipation;
+	*/
+
+	// The Secant Method, closely related to Newton's Method,
+	// is a robust approximation, approaching the zero of any curve
+	// quickly and cheaply. It does not provide infinite accuracy,
+	// but in exchange it is trivial to add more functionality to
+	// the curve it is approximating a zero of.
+	// To do so- implement your heat change in ship::DoGeneration()
+	// and then add the same change to Ship::NetIdleHeatAt().
+
+	int attempts = 10;
+	double firstGuess = 0.;
+	double firstOutput = NetIdleHeatAt(firstGuess);
+
+	double secondGuess = MaximumHeat();
+	double secondOutput = NetIdleHeatAt(secondGuess);
+
+	double middlingGuess;
+	double middlingOutput;
+
+	// Guard clause, to catch weird cases.
+	if(firstGuess == secondGuess || firstOutput <= secondOutput)
+		return secondOutput > 0. ? numeric_limits<double>::max() : 0;
+
+	// 0.001 here is a magic number which could be anything "sufficiently small".
+	// Normally it's called an Epsilon, a number that approaches but is not zero.
+	while((attempts > 0) && (fabs(firstGuess - secondGuess) > .001))
+	{
+		// Do not allow a negative-heat guess.
+		middlingGuess = max(0., firstGuess - firstOutput * (secondGuess - firstGuess) / (secondOutput - firstOutput));
+		middlingOutput = NetIdleHeatAt(middlingGuess);
+		firstGuess = secondGuess;
+		firstOutput = secondOutput;
+		secondGuess = middlingGuess;
+		secondOutput = middlingOutput;
+		--attempts;
+	}
+	// Return the most up-to-date guess.
+	return secondGuess;
+}
+
+
+
+// Get the net heat production at a certain number of heat units (not temperature).
+// If you have a % of maximum heat, multiply that by Ship::MaximumHeat() first.
+double Ship::NetIdleHeatAt(double heatLevel) const
```

Comment:
is this defined in the header? looks like a new function

Reply:
It wasn't but it is now.

Reply:
perfect